src/test/java/org/broadinstitute/barclay/argparser/TaggedArgumentTest.java

```diff
@@ -0,0 +1,301 @@
+package org.broadinstitute.barclay.argparser;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.*;
+
+/**
+ * Tests for taggable arguments, i.e.:
+ *
+ *     -I:tumor my.bam
+ *     -I:tumor,key=value,key2=value2
+ */
+public class TaggedArgumentTest {
+
+    public static class TaggableArg implements TaggedArgument {
+
+        private String tagName;
+        private Map<String, String> tagAttributes = new HashMap<>();
+        public String argValue;
+
+        public TaggableArg(final String value) {
+            this.argValue = value;
+        }
+
+        @Override
+        public void setTag(final String tagName) {
+            this.tagName = tagName;
+        }
+
+        @Override
+        public String getTag() {
+            return tagName;
+        }
+
+        @Override
+        public void setTagAttributes(final Map<String, String> attributes) {
+            this.tagAttributes.putAll(attributes);
+        }
+
+        @Override
+        public Map<String, String> getTagAttributes() {
+            return tagAttributes;
+        }
+
+        @Override
+        public String toString() { return argValue; }
+    }
+
+    @CommandLineProgramProperties(
+            summary = "Test tagged aruments",
+            oneLineSummary = "Test tagged aruments",
+            programGroup = TestProgramGroup.class
+    )
+    public class TaggableArguments {
+
+        @PositionalArguments(minElements = 0, maxElements=2)
+        public List<String> positionalArgs = new ArrayList<>();
+
+        @Argument(shortName="t", fullName="tFullName", doc="Tag target arg", optional = true)
+        public List<TaggableArg> taggableArgList = new ArrayList<>();
+
+        @Argument(shortName="s", fullName="scalarArg", optional = true)
+        public Integer taggableInteger = 17;
+
+        @Argument(shortName="n", fullName="notTaggable", optional = true)
+        public String notTaggable;
+    }
+
+    // Non-generic key-value class for easy addition to statically initialized arrays in data provider
+    private static class KeyValuePair{
```

Comment:
Can't you use `org.apache.commons.lang3.tuple.Pair`, which also has a static `of()` method?